File: fix_labels.js
const fs = require('fs');
const path = require('path');

function generateId() {
  const chars = 'abcdefghijklmnopqrstuvwxyz0123456789';
  let result = 'field-';
  for (let i = 0; i < 6; i++) {
    result += chars.charAt(Math.floor(Math.random() * chars.length));
  }
  return result;
}

const files = [
  "admin-panel/src/app/(admin)/inventory/page.tsx",
  "admin-panel/src/app/(admin)/jobs/[id]/page.tsx",
  "admin-panel/src/app/(admin)/jobs/page.tsx",
  "admin-panel/src/app/(admin)/materials/page.tsx",
  "admin-panel/src/app/(admin)/page.tsx",
  "admin-panel/src/app/(admin)/reports/page.tsx",
  "admin-panel/src/app/(admin)/sales/new/page.tsx",
  "admin-panel/src/app/(admin)/sales/page.tsx",
  "admin-panel/src/app/(admin)/staff/page.tsx",
  "admin-panel/src/app/login/page.tsx",
  "admin-panel/src/components/common/ConfirmationModal.tsx",
  "admin-panel/src/components/common/Toast.tsx",
  "admin-panel/src/components/inventory/AddStockModal.tsx",
  "admin-panel/src/components/inventory/InventoryFormModal.tsx",
  "admin-panel/src/components/layout/NotificationsDropdown.tsx",
  "admin-panel/src/components/salary/AdvanceSalaryForm.tsx",
  "admin-panel/src/components/salary/BonusForm.tsx",
  "admin-panel/src/components/salary/HolidayCalendarForm.tsx",
  "admin-panel/src/components/salary/LeaveManagement.tsx",
  "admin-panel/src/components/salary/SalaryCalculatorForm.tsx",
  "admin-panel/src/components/salary/StaffRateForm.tsx",
  "admin-panel/src/components/staff/AddStaffModal.tsx",
  "admin-panel/src/components/staff/StaffList.tsx"
];

files.forEach(filepath => {
  if (!fs.existsSync(filepath)) return;
  
  let content = fs.readFileSync(filepath, 'utf8');
  let pieces = content.split('<label');
  let new_content = pieces[0];
  
  for (let i = 1; i < pieces.length; i++) {
    let piece = pieces[i];
    if (piece.startsWith(' htmlFor') || piece.startsWith(' id')) {
      new_content += '<label' + piece;
      continue;
    }
    
    let end_idx = piece.indexOf('>');
    if (end_idx === -1) {
      new_content += '<label' + piece;
      continue;
    }
    
    let field_id = generateId();
    let new_piece = ' htmlFor="' + field_id + '"' + piece.substring(0, end_idx) + piece.substring(end_idx);
    
    let tag_match = new_piece.match(/<(input|select|textarea|Input|Select)(?=\s|>)/);
    if (tag_match) {
      let insert_pos = tag_match.index + tag_match[0].length;
      new_piece = new_piece.substring(0, insert_pos) + ' id="' + field_id + '"' + new_piece.substring(insert_pos);
    }
    
    new_content += '<label' + new_piece;
  }
  
  new_content = new_content.replace(/<(input|select|textarea|Input|Select)(?=\s|>)[^>]*>/g, (match, tag) => {
    if (match.includes('aria-label') || match.includes(' id=')) {
      return match;
    }
    let phMatch = match.match(/placeholder="([^"]+)"/);
    if (phMatch) {
      return match.replace(tag, `${tag} aria-label="${phMatch[1]}"`);
    }
    return match.replace(tag, `${tag} aria-label="Field"`);
  });
  
  if (new_content !== content) {
    fs.writeFileSync(filepath, new_content, 'utf8');
    console.log(`Fixed ${filepath}`);
  }
});

File: fix_labels.py
import os
import re
import random
import string

def generate_id():
    return 'field-' + ''.join(random.choices(string.ascii_lowercase + string.digits, k=6))

files = [
    "admin-panel/src/app/(admin)/inventory/page.tsx",
    "admin-panel/src/app/(admin)/jobs/[id]/page.tsx",
    "admin-panel/src/app/(admin)/jobs/page.tsx",
    "admin-panel/src/app/(admin)/materials/page.tsx",
    "admin-panel/src/app/(admin)/page.tsx",
    "admin-panel/src/app/(admin)/reports/page.tsx",
    "admin-panel/src/app/(admin)/sales/new/page.tsx",
    "admin-panel/src/app/(admin)/sales/page.tsx",
    "admin-panel/src/app/(admin)/staff/page.tsx",
    "admin-panel/src/app/login/page.tsx",
    "admin-panel/src/components/common/ConfirmationModal.tsx",
    "admin-panel/src/components/common/Toast.tsx",
    "admin-panel/src/components/inventory/AddStockModal.tsx",
    "admin-panel/src/components/inventory/InventoryFormModal.tsx",
    "admin-panel/src/components/layout/NotificationsDropdown.tsx",
    "admin-panel/src/components/salary/AdvanceSalaryForm.tsx",
    "admin-panel/src/components/salary/BonusForm.tsx",
    "admin-panel/src/components/salary/HolidayCalendarForm.tsx",
    "admin-panel/src/components/salary/LeaveManagement.tsx",
    "admin-panel/src/components/salary/SalaryCalculatorForm.tsx",
    "admin-panel/src/components/salary/StaffRateForm.tsx",
    "admin-panel/src/components/staff/AddStaffModal.tsx",
    "admin-panel/src/components/staff/StaffList.tsx"
]

for filepath in files:
    if not os.path.exists(filepath):
        continue
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    pieces = content.split('<label')
    new_content = pieces[0]
    
    for piece in pieces[1:]:
        if piece.startswith(' htmlFor') or piece.startswith(' id'):
            new_content += '<label' + piece
            continue
            
        end_idx = piece.find('>')
        if end_idx == -1:
            new_content += '<label' + piece
            continue
            
        field_id = generate_id()
        new_piece = f' htmlFor="{field_id}"' + piece[:end_idx] + piece[end_idx:]
        
        tag_match = re.search(r'<(input|select|textarea|Input|Select)(?=\s|>)', new_piece)
        if tag_match:
            tag_name = tag_match.group(1)
            insert_pos = tag_match.end()
            new_piece = new_piece[:insert_pos] + f' id="{field_id}"' + new_piece[insert_pos:]
            
        new_content += '<label' + new_piece
        
    def inject_aria_label(match):
        tag_content = match.group(0)
        if 'aria-label' in tag_content or ' id=' in tag_content:
            return tag_content
        ph_match = re.search(r'placeholder="([^"]+)"', tag_content)
        if ph_match:
            ph = ph_match.group(1)
            return tag_content.replace(match.group(1), f'{match.group(1)} aria-label="{ph}"', 1)
        
        return tag_content.replace(match.group(1), f'{match.group(1)} aria-label="Field"', 1)

    new_content = re.sub(r'<(input|select|textarea|Input|Select)(?=\s|>)[^>]*>', inject_aria_label, new_content)
    
    if new_content != content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Fixed {filepath}")
